streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1609,6 +1605,8 @@ public Map<String, Object> getMainConsumerConfigs(final String groupId, final St
         final Map<String, Object> mainConsumerProps = originalsWithPrefix(MAIN_CONSUMER_PREFIX);
         consumerProps.putAll(mainConsumerProps);
 
+        validateConsumerPropertyMap(consumerProps);
```

Comment:
See below: we set `group.id` and `client.id`...

We also modify `group.instance.id` -- should we log a INFO level warn for this case (a WARN is not appropriate because it's nothing wrong with user provided configs...)

We also do
```
// disable auto topic creation
consumerProps.put(ConsumerConfig.ALLOW_AUTO_CREATE_TOPICS_CONFIG, "false");
```
which I believe we can remove in here, as it's already covered vis "unmodifiable configs" now?

Reply:
Working on this change